core/src/main/java/org/elasticsearch/action/support/replication/TransportBroadcastReplicationAction.java

```diff
@@ -90,7 +91,8 @@ public void onFailure(Throwable e) {
                     int totalNumCopies = clusterState.getMetaData().index(shardId.index().getName()).getNumberOfReplicas() + 1;
                     ShardResponse shardResponse = newShardResponse();
                     ActionWriteResponse.ShardInfo.Failure[] failures;
-                    if (ExceptionsHelper.unwrap(e, UnavailableShardsException.class) != null) {
+                    if (TransportActions.isShardNotAvailableException(e) ||
```

Comment:
This feels weird to me - what would be the implication of adding UnavailableShardsException to the exceptions TransportActions.isShardNotAvailableException returns true on?